From: Hans-Joachim Picht <hpicht@redhat.com>
Date: Mon, 10 Dec 2007 14:03:12 +0100
Subject: [s390] pav alias disks not detected on lpar
Message-id: 20071210130312.GA29540@redhat.com
O-Subject: [RHEL5 U2 PATCH 1/8] s390 - pav alias disks not detected on lpar
Bugzilla: 416081
RH-Acked-by: Pete Zaitcev <zaitcev@redhat.com>

Description
============

Modify the sense id channel program to allow device sensing of pav
alias devices which belong to a base device with ungrouped paths.

Bugzilla
=========

BZ 416081
https://bugzilla.redhat.com/show_bug.cgi?id=416081

Upstream status of the patch:
=============================
Patch included in git as commit c94dec99f9759c41cadf0f2781846da5b40a98f6

Test status:
============
Kernel with patch was built and successfully tested

Please ACK.

With best regards,

Hans

diff --git a/drivers/s390/cio/device_id.c b/drivers/s390/cio/device_id.c
index f3cc82a..fdcc07f 100644
--- a/drivers/s390/cio/device_id.c
+++ b/drivers/s390/cio/device_id.c
@@ -117,14 +117,6 @@ __ccw_device_sense_id_start(struct ccw_device *cdev)
 	sch = to_subchannel(cdev->dev.parent);
 	/* Setup sense channel program. */
 	ccw = cdev->private->iccws;
-	if (sch->schib.pmcw.pim != 0x80) {
-		/* more than one path installed. */
-		ccw->cmd_code = CCW_CMD_SUSPEND_RECONN;
-		ccw->cda = 0;
-		ccw->count = 0;
-		ccw->flags = CCW_FLAG_SLI | CCW_FLAG_CC;
-		ccw++;
-	}
 	ccw->cmd_code = CCW_CMD_SENSE_ID;
 	ccw->cda = (__u32) __pa (&cdev->private->senseid);
 	ccw->count = sizeof (struct senseid);
